sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -99,29 +105,30 @@ FileBodyStream::FileBodyStream(const std::string& filename)
                                    // intended to be sequential from beginning to end.
         NULL);
 #else
-    m_filehandle = CreateFile2(
+    fileHandle = CreateFile2(
         std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>>().from_bytes(filename).c_str(),
         GENERIC_READ,
         FILE_SHARE_READ,
         OPEN_EXISTING,
         NULL);
 #endif
 
-    if (m_filehandle == INVALID_HANDLE_VALUE)
+    if (fileHandle == INVALID_HANDLE_VALUE)
     {
       throw std::runtime_error("Failed to open file for reading. File name: '" + filename + "'");
     }
     LARGE_INTEGER fileSize;
-    if (!GetFileSizeEx(m_filehandle, &fileSize))
+    if (!GetFileSizeEx(fileHandle, &fileSize))
     {
       throw std::runtime_error("Failed to get size of file. File name: '" + filename + "'");
     }
+    m_filehandle = static_cast<void*>(fileHandle);
     m_randomAccessFileBodyStream = std::make_unique<_internal::RandomAccessFileBodyStream>(
         _internal::RandomAccessFileBodyStream(m_filehandle, 0, fileSize.QuadPart));
   }
   catch (std::exception&)
   {
-    CloseHandle(m_filehandle);
+    CloseHandle(fileHandle);
```

Comment:
The handle should be in a holder at the point of initialization.  That way we don't need to setup a try/catch in this method.

Reply:
That is unrelated to this PR and its intended change. The issue already exists in master.

I would have filed an issue, but this was discussed previously, and after extensive discussions with Jeff, we decided to avoid the "holder" pattern.
For example: https://github.com/Azure/azure-sdk-for-cpp/pull/1830#discussion_r591891880

Reply:
Please file an issue, we should fix the code to use the holder pattern to avoid unnessecary handler creation and to ensure release.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/2199